prime-angular-frontend/src/app/modules/enrolment/pages/absence-management-page/absence-management-page.component.ts

```diff
@@ -43,17 +43,33 @@ export class AbsenceManagementPageComponent extends AbstractEnrolmentPage implem
   }
 
   public isCurrent(): boolean {
-    return moment().isBetween(this.absence?.startTimestamp, this.absence?.endTimestamp);
+    return moment().isAfter(this.absence?.startTimestamp);
   }
 
   public endAbsence(): void {
     this.busy = this.enrolmentResource.endCurrentEnrolleeAbsence(this.enrolmentService.enrolment.id)
-      .subscribe(() => this.absence = null);
+      .subscribe(() => {
+        this.absence = null;
+        this.formState.email.removeValidators(Validators.required);
+      });
   }
 
   public cancelAbsence(absenceId: number): void {
     this.busy = this.enrolmentResource.deleteFutureEnrolleeAbsence(this.enrolmentService.enrolment.id, absenceId)
-      .subscribe(() => this.absence = null);
+      .subscribe(() => {
+        this.absence = null;
+        this.formState.email.removeValidators(Validators.required);
+      });
+  }
+
+  public sendEmail(): void {
+    this.formState.email.markAsTouched();
+    if (!this.formState.email.valid) {
```

Comment:
For readability we can use `if(this.formState.email.invalid)`

Reply:
FormControl's can be in 4 states
https://angular.io/api/forms/FormControlStatus

So when checking if something is !valid it is actually different than checking if it is invalid. This is why we want to fail out if the form control is !valid